modules/bash-commons/src/aws.sh

```diff
@@ -20,36 +20,65 @@ readonly three_hours_in_s=10800
 # a token from IMDSv2
 readonly six_hours_in_s=21600
 
-# Detect if the instance is using IMDSv2 or if it is using IMDSv1 still.
+# shellcheck source=./modules/bash-commons/src/assert.sh
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/assert.sh"
+# shellcheck source=./modules/bash-commons/src/log.sh
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/log.sh"
+
+# Set variable to an empty string if it is unbound to prevent "unbound variable".
+export GRUNTWORK_BASH_COMMONS_IMDS_VERSION=${GRUNTWORK_BASH_COMMONS_IMDS_VERSION-}
+# Set default variable to version "2" if nothing is set.
+export default_instance_metadata_version=${default_instance_metadata_version-"2"}
+
+# Detect if the instance has IMDS available and return what version is available.
 # Users can always specify the version of the Instance Metadata Service they want bash-commons
-# to consult by setting the environment variable GRUNTWORK_BASH_COMMONS_IMDS_VERSION
-# Defaults to IMDSv2 since that is now enabled by default on instances (IMDS only has two options,
-# "optional" = both v1 and v2, or "required" = v2 only).  All new instances support v2 now.
-if [[ -z "$GRUNTWORK_BASH_COMMONS_IMDS_VERSION" ]]; then
-    to_token_or_not_to_token=$(sudo curl -s -o /dev/null -X PUT ${imdsv2_token_endpoint} -H "X-aws-ec2-metadata-token-ttl-seconds: 10"; echo $?)
-    if [ ${to_token_or_not_to_token} -eq 0 ]; then
+# to consult by setting the environment variable GRUNTWORK_BASH_COMMONS_IMDS_VERSION.
+# If set, GRUNTWORK_BASH_COMMONS_IMDS_VERSION will override default_instance_metadata_version.
+# Defaults to IMDSv2 since that is now available by default on instances.
+function aws_check_metadata_availability {
```

Comment:
I saw in the docs, it is preferred to make this a function so that other scripts/modules can call it.